docs/services/keepalived.md

```diff
@@ -0,0 +1,51 @@
+
+#### How does keepalived works
+
+[Keepalived](https://www.google.com/url?q=https://www.keepalived.org/doc/index.html&sa=D&ust=1568617551393000&usg=AFQjCNEhOHfFQqFda0lef-rtR8U51QFS3A) is used for IP failover between two servers. Its facilities for load balancing and high-availability to Linux-based infrastructures.Keepalived works on VRRP (Virtual Router Redundancy Protocol) protocol to make IPs highly available .
+
+The VRRP protocol ensures that one of participating nodes is master. The backup node(s) listens for multicast packets from a node with a higher priority. If the backup node fails to receive VRRP advertisements for a period longer than three times of the advertisement timer, the backup node takes the master state and assigns the configured IP(s) to itself. In case there are more than one backup nodes with the same priority, the one with the highest IP wins the election.
+
+
+Note:-No fencing mechanism available in keepalived. If  two participating nodes don't see each other, both will have the master state and both will carry the same IP(s)
+
+
+
+
+In our Current infrastructure we are using it for couchdb proxy failover in ICDS environment.
+For more info see here
+For a sample config see here
+We have plans to implement it more places where we have proxies setup.
+
+
+#####Operational Notes:-
+
+1. check keepalived status
+    ```# systemctl status keepalived```
+
+2. know which one is the master
+      Check the status/logs and you’ll see the log lines like this
+     ```Sep 12 03:25:36 MUMGCCWCDPRDCDV09 Keepalived_vrrp[30570]: VRRP_Instance(VI_1) Entering BACKUP STATE```
+     Check the virtual ip listed in /etc/keepalived/keepalived.conf,verify if this ip address is assigned to the interface of the server.
+
+3. where are the logs
+    Keepalived logs to journald 
+    ```# journalctl -u keepalived```
+
+4. if we restart haproxy will that trigger a failover? ( only on master node)
+    ```
+    vrrp_script chk_service {           
+        script "pgrep haproxy"  
+        interval 2    
+      }```
+
+From the config above it check for running process in every two second interval.if restart took longer than this , it will trigger failover
```

Comment:
```suggestion
From the config above it check for running process in every two second interval.if restart took longer than this, it will trigger failover
```